integration_tests/utils.py

```diff
@@ -21,3 +61,197 @@ def assert_click_command(command, args):
     print(result.output)
     assert not result.exception
     assert result.exit_code == 0
+
+
+def create_empty_geotiff(path):
+    # Example method, not used
+    metadata = {'count': 1,
+                'crs': 'EPSG:28355',
+                'driver': 'GTiff',
+                'dtype': 'int16',
+                'height': 8521,
+                'nodata': -999.0,
+                'transform': [25.0, 0.0, 638000.0, 0.0, -25.0, 6276000.0],
+                'compress': 'lzw',
+                'width': 9721}
+    with rasterio.open(path, 'w', **metadata) as dst:
+        pass
+
+
+def limit_num_measurements(dataset_type):
+    if 'measurements' not in dataset_type:
+        return
+    measurements = dataset_type['measurements']
+    if len(measurements) > TEST_STORAGE_NUM_MEASUREMENTS:
+        dataset_type['measurements'] = measurements[:TEST_STORAGE_NUM_MEASUREMENTS]
+    return dataset_type
+
+
+def prepare_test_ingestion_configuration(tmpdir,
+                                         output_dir,
+                                         filename,
+                                         mode=None):
+    customizers = {
+        'fast_ingest': edit_for_fast_ingest,
+        'end2end': edit_for_end2end,
+    }
+
+    filename = Path(filename)
+    if output_dir is None:
+        output_dir = tmpdir.mkdir(filename.stem)
+    config = load_yaml_file(filename)[0]
+
+    if mode is not None:
+        if mode not in customizers:
+            raise ValueError('Wrong mode: ' + mode)
+        config = customizers[mode](config)
+
+    config['location'] = str(output_dir)
+
+    # If ingesting with the s3test driver
+    if 'bucket' in config['storage']:
+        config['storage']['bucket'] = str(output_dir)
+
+    config_path = tmpdir.join(filename.name)
+    with open(str(config_path), 'w') as stream:
+        yaml.dump(config, stream)
```

Comment:
I know this was just moved, but, we should probably use `safe_dump`, cause we use `safe_load` and the two might not be compatible for things like tuples, datetimes etc.

Reply:
Good catch, thanks.